platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -58,19 +53,16 @@
      *
      * @param configurationDirectoryPath directory to watch for changes
      * @param fileExtension              extension of of the files to watch
-     * @param watchService               watch service to use
      * @param executorService            executor service used to create the watch thread
      */
     public ConfigurationFilesPoller(@NotNull Path configurationDirectoryPath,
-            @NotNull String fileExtension, @NotNull WatchService watchService,
-            @NotNull ExecutorService executorService) {
+            @NotNull String fileExtension, @NotNull ExecutorService executorService) {
         notNull(configurationDirectoryPath, "configurationDirectoryPath cannot be null");
         notNull(fileExtension, "fileExtension cannot be null");
-        notNull(watchService, "watchService cannot be null");
         notNull(executorService, "executorService cannot be null");
 
         this.configurationDirectoryPath = configurationDirectoryPath;
-        this.watchService = watchService;
+        this.watchService = new FileAlterationObserver(configurationDirectoryPath.toString());
```

Comment:
❓ Do you think we could have used the `FileAlterationMonitor` to replace this whole class?